pydis_site/apps/content/tests/helpers.py

```diff
@@ -75,29 +75,32 @@ class MockPagesTestCase(fake_filesystem_unittest.TestCase):
-        self.setUpPyfakefs()
-
-        self.fs.create_file(f"{BASE_PATH}/_info.yml", contents=CATEGORY_INFO)
-        self.fs.create_file(f"{BASE_PATH}/root.md", contents=MARKDOWN_WITH_METADATA)
-        self.fs.create_file(
-            f"{BASE_PATH}/root_without_metadata.md", contents=MARKDOWN_WITHOUT_METADATA
-        )
-        self.fs.create_file(f"{BASE_PATH}/not_a_page.md/_info.yml", contents=CATEGORY_INFO)
-        self.fs.create_file(f"{BASE_PATH}/category/_info.yml", contents=CATEGORY_INFO)
-        self.fs.create_file(
-            f"{BASE_PATH}/category/with_metadata.md", contents=MARKDOWN_WITH_METADATA
-        )
-        self.fs.create_file(f"{BASE_PATH}/category/subcategory/_info.yml", contents=CATEGORY_INFO)
-        self.fs.create_file(
-            f"{BASE_PATH}/category/subcategory/with_metadata.md", contents=MARKDOWN_WITH_METADATA
-        )
-        self.fs.create_file(
-            f"{BASE_PATH}/category/subcategory/without_metadata.md",
-            contents=MARKDOWN_WITHOUT_METADATA
-        )
+        Path(f"{BASE_PATH}/_info.yml").write_text(CATEGORY_INFO)
+        Path(f"{BASE_PATH}/root.md").write_text(MARKDOWN_WITH_METADATA)
+        Path(f"{BASE_PATH}/root_without_metadata.md").write_text(MARKDOWN_WITHOUT_METADATA)
+        Path(f"{BASE_PATH}/not_a_page.md").mkdir(exist_ok=True)
+        Path(f"{BASE_PATH}/not_a_page.md/_info.yml").write_text(CATEGORY_INFO)
+        Path(f"{BASE_PATH}/category").mkdir(exist_ok=True)
+        Path(f"{BASE_PATH}/category/_info.yml").write_text(CATEGORY_INFO)
+        Path(f"{BASE_PATH}/category/with_metadata.md").write_text(MARKDOWN_WITH_METADATA)
+        Path(f"{BASE_PATH}/category/subcategory").mkdir(exist_ok=True)
+        Path(f"{BASE_PATH}/category/subcategory/_info.yml").write_text(CATEGORY_INFO)
+        Path(
+            f"{BASE_PATH}/category/subcategory/with_metadata.md"
+        ).write_text(MARKDOWN_WITH_METADATA)
+        Path(
+            f"{BASE_PATH}/category/subcategory/without_metadata.md"
+        ).write_text(MARKDOWN_WITHOUT_METADATA)
 
         temp = f"{BASE_PATH}/tmp"  # noqa: S108
-        self.fs.create_file(f"{temp}/_info.yml", contents=CATEGORY_INFO)
-        self.fs.create_file(f"{temp}.md", contents=MARKDOWN_WITH_METADATA)
-        self.fs.create_file(f"{temp}/category/_info.yml", contents=CATEGORY_INFO)
-        self.fs.create_dir(f"{temp}/category/subcategory_without_info")
+        Path(f"{temp}").mkdir(exist_ok=True)
+        Path(f"{temp}/_info.yml").write_text(CATEGORY_INFO)
+        Path(f"{temp}.md").write_text(MARKDOWN_WITH_METADATA)
+        Path(f"{temp}/category").mkdir(exist_ok=True)
+        Path(f"{temp}/category/_info.yml").write_text(CATEGORY_INFO)
+        Path(f"{temp}/category/subcategory_without_info").mkdir(exist_ok=True)
+
+    @classmethod
+    def setUpTestData(cls):
+        # May get called multiple times - ignore erorrs in that case.
+        atexit.register(shutil.rmtree, BASE_PATH, ignore_errors=True)
```

Comment:
Why not put this atexit register right after the directory is created. This would ensure it is run only once, and also run even if the tests weren't run for some reason (e.g. selectively running tests possibly?).

Alternatively it may make more sense to move the tempdir to the test setup function, so it is only called when the tests are run. (could also use a `tempfile.TemporaryDirectory` which has a `cleanup` method instead of needing to rmtree but should be the same so doesn't matter).

Reply:
This is a good shout. I have implemented it in 88157999.

Reply:
As for having the tempdir in the test setup function, I do think it's a good idea, but it would require a larger refactoring due to the places BASE_PATH is referenced.